discord/ext/menus/__init__.py

```diff
@@ -898,17 +984,18 @@ class MenuPages(Menu):
         The current page that we are in. Zero-indexed
         between [0, :attr:`PageSource.max_pages`).
     """
-    def __init__(self, source, **kwargs):
+
+    def __init__(self, source: PageSource[T], **kwargs: Any):
         self._source = source
-        self.current_page = 0
+        self.current_page: int = 0
         super().__init__(**kwargs)
 
     @property
-    def source(self):
+    def source(self) -> PageSource:
```

Comment:
Shouldn't this return PageSource[T]
```suggestion
    def source(self) -> PageSource[T]:
```